### PR TITLE
carriage return on the first prompt to properly capture all input

### PR DESCRIPTION
**Pull request by scopatz**
This should address #959 for both ptk and readline.

### CODE REVIEW

File: xonsh/ptk/shell.py [hunk 1]
```diff
@@ -169,6 +170,9 @@ def prompt_tokens(self, cli):
         except Exception:  # pylint: disable=broad-except
             print_exception()
         toks = partial_color_tokenize(p)
+        if self._first_prompt:
+            carriage_return()
+            self._first_message = False
```
Comment (gforsyth) [inline xonsh/ptk/shell.py]: should this be `self._first_prompt`?

**Comment by gforsyth** [comment]
Hey @scopatz -- seems to work on PTK! I've never had this problem with readline so can't judge.

File: xonsh/ptk/shell.py (hunk 1 above)
Comment (scopatz) [inline xonsh/ptk/shell.py]: Nice, thanks, it should be!

**Comment by scopatz** [comment]
Ok the fix is in. I was seeing this problem on both ptk and readline, and this branch does fix both.

**Comment by scopatz** [comment]
Also, this was somewhat tricky to figure out given how easy the solution is.

**Comment by gforsyth** [comment]
> Also, this was somewhat tricky to figure out given how easy the solution is.

That's always the way of these things, isn't it?

**Comment by scopatz** [comment]
> That's always the way of these things, isn't it?

Its it true! Especially for our oldest bugs....

**Comment by melund** [comment]
Lgtm

**Comment by scopatz** [comment]
Thanks @melund!

**Pull request merged**